x-pack/plugins/cloud_security_posture/public/components/setup_status.tsx

```diff
@@ -0,0 +1,187 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import type { UseQueryResult } from 'react-query';
+import { EuiEmptyPrompt } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { NoDataPage } from '@kbn/kibana-react-plugin/public';
+import { css } from '@emotion/react';
+import { CspLoadingState } from './csp_loading_state';
+import { useCisKubernetesIntegration } from '../common/api/use_cis_kubernetes_integration';
+import { useCISIntegrationLink } from '../common/navigation/use_navigate_to_cis_integration';
+
+export const LOADING_STATE_TEST_SUBJECT = 'csp_setup_status_loading';
+export const ERROR_STATE_TEST_SUBJECT = 'csp_setup_status_error';
+export const PACKAGE_NOT_INSTALLED_TEST_SUBJECT = 'csp_setup_status_package_not_installed';
+export const DEFAULT_NO_DATA_TEST_SUBJECT = 'csp_setup_status_no_data';
+
+interface CommonError {
+  body: {
+    error: string;
+    message: string;
+    statusCode: number;
+  };
+}
+
+export const isCommonError = (error: any): error is CommonError => {
+  if (!error?.body) return false;
+
+  const {
+    body: { error: innerError, message, statusCode },
+  } = error;
+
+  return !!(innerError && message && statusCode);
+};
+
+const packageNotInstalledRenderer = (cisIntegrationLink?: string) => (
+  <NoDataPage
+    data-test-subj={PACKAGE_NOT_INSTALLED_TEST_SUBJECT}
+    css={css`
+      max-width: 950px;
+      margin-top: 50px;
+      margin-left: auto;
+      margin-right: auto;
+    `}
+    pageTitle={i18n.translate('xpack.csp.setupStatus.packageNotInstalled.pageTitle', {
+      defaultMessage: 'Install Integration to get started',
+    })}
+    solution={i18n.translate('xpack.csp.setupStatus.packageNotInstalled.solutionNameLabel', {
+      defaultMessage: 'Cloud Security Posture',
+    })}
+    // TODO: Add real docs link once we have it
+    docsLink={'https://www.elastic.co/guide/index.html'}
+    logo={'logoSecurity'}
+    actions={{
+      elasticAgent: {
+        href: cisIntegrationLink,
+        isDisabled: !cisIntegrationLink,
+        title: i18n.translate('xpack.csp.setupStatus.packageNotInstalled.buttonLabel', {
+          defaultMessage: 'Add a CIS integration',
+        }),
+        description: i18n.translate('xpack.csp.setupStatus.packageNotInstalled.description', {
+          defaultMessage:
+            'Use our CIS Kubernetes Benchmark integration to measure your Kubernetes cluster setup against the CIS recommendations.',
+        }),
+      },
+    }}
+  />
+);
+
+const defaultLoadingRenderer = () => (
+  <CspLoadingState data-test-subj={LOADING_STATE_TEST_SUBJECT}>
+    <FormattedMessage id="xpack.csp.setupStatus.loadingDescription" defaultMessage="Loading..." />
+  </CspLoadingState>
+);
+
+const defaultErrorRenderer = (error: unknown) => (
+  <EuiEmptyPrompt
+    css={css`
+      margin-top: 50px;
+    `}
```

Comment:
managed to center this error message by using 
```
css={css`
      position: absolute;
      inset: 0;
      height: fit-content;
    `}
```

not sure if its a good enough solution for both cases, but it might be a direction. as you said we depends on the parent here, and the bottom line is that we need it to take 100% height. 

![image](https://user-images.githubusercontent.com/51442161/178482163-97559d7d-8b38-400e-a809-1c077897c5f8.png)

Reply:
I'll handle centering when we'll only be rendered in the security solution in https://github.com/elastic/kibana/issues/134425